src/AutoRest.CSharp/LowLevel/Generation/LowLevelModelWriter.cs

```diff
@@ -54,7 +55,7 @@ private static void WriteConstructor(CodeWriter writer, ConstructorSignature sig
                 {
                     var field = model.GetFieldByParameterName(parameter.Name);
                     writer
-                        .Append($"{field.Name:I} = {parameter.Name:I}")
+                        .Append($"{field.Declaration.RequestedName:I} = {parameter.Name:I}")
```

Comment:
`field.Name => field.Declaration.Name` which must be initialized by printing `field.Declaration` through `writer`. See the codes here: https://github.com/Azure/autorest.csharp/blob/722b417ce6c210ff8daaf2149f82a8cc98cb83a8/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriter.cs#L117

That means we must `writeFields()` first, otherwise `field.Declaration.Name` is not initialized. To avoid this prerequisite, we can use `field.Declaration.RequestName` since property names should be unique.

Reply:
Resolved by using `CodeWriterScopeDeclaration`